processor/src/main/java/com/linecorp/decaton/processor/runtime/ProcessorSubscription.java

```diff
@@ -266,9 +267,15 @@ public void initiateShutdown() {
     }
 
     @Override
-    public void awaitShutdown() throws InterruptedException {
-        join();
+    public boolean awaitShutdown(Duration limit) throws InterruptedException {
+        join(limit.toMillis());
         metrics.close();
```

Comment:
Hmm. I think the right thing is for the thread to call `metrics.close()` when it's finished everything else, since `initiateShutdown()` should cause the shutdown to happen and `awaitShutdown` should observe that process.